core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -376,12 +381,36 @@ class Partition(val topic: String,
   }
 
   /**
+   * Check and maybe increment the low watermark of the partition;
+   * this function can be triggered when any replica's logStartOffset changed
```

Comment:
If LW is only used for DeleteRecordResponse, do we need to store and maintain it in Replica? Could we just compute it on the fly when serving DeleteRecordRequests?